src/platform_impl/linux/system_tray.rs

```diff
@@ -102,10 +102,24 @@ impl Drop for SystemTray {
 }
 
 fn temp_icon_path() -> std::io::Result<(PathBuf, PathBuf)> {
-  let mut parent_path = std::env::temp_dir();
+  let mut parent_path = dirs_next::runtime_dir().unwrap_or_else(|| std::env::temp_dir());
+
   parent_path.push("tao");
   std::fs::create_dir_all(&parent_path)?;
   let mut icon_path = parent_path.clone();
   icon_path.push(format!("tray-icon-{}.png", uuid::Uuid::new_v4()));
   Ok((parent_path, icon_path))
 }
+
+#[test]
+fn temp_icon_path_prefers_runtime() {
+  let runtime_dir = env!("XDG_RUNTIME_DIR");
```

Comment:
this test will fail if you don't have xdg dirs set up FYI

Reply:
Nice catch, will push a commit to use `option_env!` asap

Reply:
Ha, yes you're right. Though the test itself shows it would at runtime fall back on `/tmp` for those users :sweat_smile:
Should it have been an early return?